### PR TITLE
Implement exercise - collatz-conjecture

### PR DESCRIPTION
**Pull request by marvelou-s**
Hi there! I've brought new exercise [collatz-conjecture](https://github.com/exercism/problem-specifications/tree/master/exercises/collatz-conjecture) ;)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -567,6 +567,17 @@
         "classes",
         "data_structure"
       ]
+    },
+    {
+      "slug": "collatz-conjecture",
+      "uuid": "c74f60b3-48fe-46c5-ae52-5e4b0787ba84",
+      "core": false,
+      "unlocked_by": null,
```
Comment (patricksjackson) [inline config.json]: Let's have this unlocked by the leap exercise.

**Comment by KevinWMatthews** [comment]
@patricksjackson , I've merged the C++14 changes from #292 into the current master branch.

I merged the new `master` into this branch so that `implement_collatz_conjecture` has the latest C++14 Travis settings, then updated this exercise's `CMakeLists.txt` to use C++14. I think this is the way to go.

It's probably best to leave out the sanitizer changes in #293 for now. I have a few more experiments to run before we merge those changes.

@marvelou-s I've pushed these changes to this branch. Be sure to pull (your own branch!) if you do more work. Thanks for your contribution!

File: exercises/collatz-conjecture/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+if(EXERCISM_COMMON_CATCH)
+    # For Exercism track development only
+    add_executable(${exercise} ${file}_test.cpp ${exercise_cpp} ${file}.h $<TARGET_OBJECTS:catchlib>)
+else()
+    # Build executable from sources and headers
+    add_executable(${exercise} ${file}_test.cpp ${exercise_cpp} ${file}.h test/tests-main.cpp)
+endif()
+
+set_target_properties(${exercise} PROPERTIES
+    CXX_STANDARD 14
```
Comment (KevinWMatthews) [inline exercises/collatz-conjecture/CMakeLists.txt]: C++ standard updated to `C++14` per latest `master` branch.

**Comment by patricksjackson** [comment]
@iHiD We merged a new exercise, I think you still need to kick off an update script manually?

@marvelou-s Thanks!

**Comment by iHiD** [comment]
Done. Thanks :)

**Pull request merged**